sdk/assertion.go

```diff
@@ -123,6 +123,32 @@ func (a Assertion) GetHash() ([]byte, error) {
 	return ocrypto.SHA256AsHex(transformedJSON), nil
 }
 
+func (s *Statement) MarshalJSON() ([]byte, error) {
+	// Define a custom struct for serialization
+	type Alias Statement
+	aux := &struct {
+		Value interface{} `json:"value,omitempty"`
+		*Alias
+	}{
+		Alias: (*Alias)(s),
+	}
+
+	if s.Format == "json-structured" {
+		raw := json.RawMessage(s.Value)
+		var tmp interface{}
+		// Attempt to decode Value to validate it's actual json
+		if err := json.Unmarshal(raw, &tmp); err == nil {
+			aux.Value = raw
+		} else {
+			aux.Value = s.Value
+		}
+	} else {
+		aux.Value = s.Value
+	}
```

Comment:
I reviewed our implementation of assertions compared to our spec, and it looks like we missed the enum of the `statement.format`. I've raised this with our architects, and we're going to review the alignment of the ZTDF/BaseTDF spec with the official SDKs (Go, Java, and Web).

I'd suggest implementing the marshaling within your PEP to support this in the short term. Something like:

```go
type MyStatement struct {
  opentdf.Statement
}

func (s *MyStatement) MarshalJSON() ([]byte, error) {
  // ...
}
```

Reply:
Alternatively you should be able to just marshal the data since we have the json tags on the `Statement`.

Reply:
This doesn't work.

`AssertionConfig.Statement` would need to be an interface, so I could have a custom `MarshalJSON` implementation be called.
I cannot assign `MyStatement` to `AssertionConfig` - if I wrap that in `MyAssertionConfig`and adjust the statement type there, I cannot use it in `WithAssertions` .. and I cannot add it to TDFConfig so it's not used in `CreateTDFContext`.

I don't think it makes sense to go down this road